tasks/main.yml

```diff
@@ -28,6 +28,12 @@
   when:
     - go
 
+- name: Build Go project if necessary
+  ansible.builtin.include_tasks: build_go_tool.yml
+  when:
+    - go
+    - go_build
```

Comment:
Is it really necessary to include `- go` in this `when` block?  Or does this tie into why you made the `go_build` default to `true`?

Reply:
Yes.  If the Go project doesn't do things in the normal way then you might need to install the Go toolchain but manually install the Go dependencies and build the project.  This would be analogous to a situation where a poorly written Python project might state what Debian packages to install in the `README.md` instead of providing a `setup.py` or `requirements.txt` file.

Reply:
Thanks for the explanation! 👍